### PR TITLE
Enable test: discarding task group making sure we deinit confidently

### PR DESCRIPTION
**Pull request by ktoso**
Similar to https://github.com/apple/swift/pull/67392 / rdar://110474311 we believe this type of failure: 

```
Input was:
<<<<<<
          1: deinit main.ErrorFirst 
check:72                      X~~~~~ error: no match found
>>>>>>
```

to be caused by the process exiting before the deinits complete running. We add a latch to make sure the deinits got to run before we exit the app

---

Resolves rdar://110025115 and re-enables `Concurrency/Runtime/async_taskgroup_discarding.swift`

### CODE REVIEW

**Review by drexin — APPROVED** [review]
LGTM

File: test/Concurrency/Runtime/async_taskgroup_discarding.swift [hunk 1]
```diff
@@ -19,29 +16,76 @@ import _Concurrency
...
+  }
+
+  func hit() {
+    defer { count -= 1 }
+    print("hit @ \(count)")
+    if count == 0 {
+      fatalError("Counted down more times than expected! (From: \(from))")
+    } else if count == 1 {
+      print("hit resume")
+      print("resume")
```
Comment (tomerd) [inline test/Concurrency/Runtime/async_taskgroup_discarding.swift:37]: second print redundant?

**Review by tomerd — APPROVED** [review]

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: test/Concurrency/Runtime/async_taskgroup_discarding.swift (hunk 1 above)
Comment (ktoso) [reply test/Concurrency/Runtime/async_taskgroup_discarding.swift:37]: yep thanks

Comment (ktoso) [reply test/Concurrency/Runtime/async_taskgroup_discarding.swift:37]:
```suggestion
```

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Pull request merged**